x-pack/plugins/viz_editor/common/query_types.ts

```diff
@@ -8,13 +8,24 @@
 // Select clause
 // ----------------------------------------
 
+export type OperationName = 'column' | 'count' | 'date_histogram' | 'sum' | 'avg';
+
 export interface Aliasable {
+  operation: OperationName;
   alias?: string;
 }
 
-export interface Column extends Aliasable {
-  operation: 'col';
-  argument: string;
+export interface GenericOperation extends Aliasable {
+  argument: Field;
+}
+
+export interface Field {
+  field: string;
+}
+
+export interface ColumnOperation extends Aliasable {
+  operation: 'column';
```

Comment:
And maybe this should be `operation: 'field'` instead of `operation: 'column'`? Dunnoes.